etna/datasets/internal_datasets.py

```diff
@@ -500,6 +498,9 @@ def get_m3_dataset(dataset_dir: Path, dataset_freq: str) -> None:
         df_full_exog = pd.concat([df_full_exog, df_full_part_exog])
         df_test_exog = pd.concat([df_test_exog, df_test_part_exog])
 
+    df_full_exog["origin_timestamp"] = pd.to_datetime(df_full_exog["origin_timestamp"])
```

Comment:
As I understand, during reading `csv` it will have no effect, it will read `object`.